### PR TITLE
#149 v2 hive db

### PR DESCRIPTION
**Pull request by peter-mcclonski**

### CODE REVIEW

File: foundation/foundation-mda/src/main/resources/profiles.json [hunk 1]
```diff
@@ -989,29 +992,6 @@
 			}
 		]
 	},
-	{
-		"name": "aissemble-hive-metastore-db-kubernetes",
```
Comment (peter-mcclonski) [inline foundation/foundation-mda/src/main/resources/profiles.json:993]: I: Deleting unused profiles

File: foundation/foundation-mda/src/main/resources/templates/deployment/spark-infrastructure/v2/spark.infrastructure.values.dev.yaml.vm [hunk 2]
```diff
@@ -6,3 +6,12 @@ aissemble-spark-history-chart:
 
     ingress:
       enabled: false
+
+aissemble-hive-metastore-service-chart:
```
Comment (peter-mcclonski) [inline foundation/foundation-mda/src/main/resources/templates/deployment/spark-infrastructure/v2/spark.infrastructure.values.dev.yaml.vm:10]: I: We don't want to persist between tests for local dev

File: foundation/foundation-mda/src/main/resources/templates/deployment/spark-infrastructure/v2/spark.infrastructure.values.yaml.vm [hunk 3]
```diff
@@ -42,6 +42,16 @@ aissemble-thrift-server-chart:
      </configuration>
 
 aissemble-hive-metastore-service-chart:
+  mysql:
+    auth:
+      # Note: Changing these values requires removal of the `hive-metastore-db-0` PVC, or manual modification of the
+      # persisted database.
+      username: hive
+      # NB: UCS
```
Comment (peter-mcclonski) [inline foundation/foundation-mda/src/main/resources/templates/deployment/spark-infrastructure/v2/spark.infrastructure.values.yaml.vm:50]: I: Flagging for future improvement as we roll out the UCS.  This chart will not be active by default until improved via UCS.

File: foundation/foundation-mda/src/main/resources/templates/deployment/spark-infrastructure/v2/spark.infrastructure.values.dev.yaml.vm (hunk 2 above)
Comment (carter-cundiff) [reply foundation/foundation-mda/src/main/resources/templates/deployment/spark-infrastructure/v2/spark.infrastructure.values.dev.yaml.vm:10]: Q: What do you mean by this? What is primary and secondary in this context?

**Comment by carter-cundiff** [comment]
Q: Do we need to add this chart to `foundation/foundation-mda/src/main/resources/templates/deployment/spark-infrastructure/v2/spark.infrastructure.chart.yaml.vm`?

File: foundation/foundation-mda/src/main/resources/profiles.json (hunk 1 above)
Comment (carter-cundiff) [reply foundation/foundation-mda/src/main/resources/profiles.json:993]: Q: Did this result in an update to the generator comments somewhere for these files? Not seeing it in the PR.

Comment (carter-cundiff) [reply foundation/foundation-mda/src/main/resources/profiles.json:993]: Why not? The underlying file has been deleted.

File: foundation/foundation-mda/src/main/resources/templates/deployment/spark-infrastructure/v2/spark.infrastructure.values.dev.yaml.vm (hunk 2 above)
Comment (peter-mcclonski) [reply foundation/foundation-mda/src/main/resources/templates/deployment/spark-infrastructure/v2/spark.infrastructure.values.dev.yaml.vm:10]: https://dev.mysql.com/doc/refman/5.7/en/group-replication-primary-secondary-replication.html#:~:text=There%20is%20a%20primary%20(source,in%20row%2Dbased%20replication).

Basically it's MySQL terminology for replicas.  If this is set to `true`, then a pvc will be created which persists the entire database between restarts unless you either manually delete it or clear your Kubernetes workloads.  For local development, this is generally not desirable-- so we avoid the behavior, much as we do for Kafka in local dev.

File: extensions/extensions-helm/extensions-helm-spark-infrastructure/aissemble-hive-metastore-service-chart/values.yaml [hunk 4]
```diff
@@ -14,6 +14,13 @@ image:
   imagePullPolicy: IfNotPresent
   dockerRepo: "ghcr.io/"
 
+mysql:
+  enabled: true
+  fullnameOverride: "hive-metastore-db"
+  auth:
+    # The schematool for the metastore service will create the database for us.
+    database: "metastore"
```
[the remark this answers is not in the document]
Comment (peter-mcclonski) [reply extensions/extensions-helm/extensions-helm-spark-infrastructure/aissemble-hive-metastore-service-chart/values.yaml:22]: Done, good catch.

File: foundation/foundation-mda/src/main/resources/profiles.json (hunk 1 above)
Comment (peter-mcclonski) [reply foundation/foundation-mda/src/main/resources/profiles.json:993]: That's actually a fair point.  I'll go ahead and delete them.

Comment (peter-mcclonski) [reply foundation/foundation-mda/src/main/resources/profiles.json:993]: Done

**Review by carter-cundiff — APPROVED** [review]

**Pull request merged**